src/Handlers/ResultHandler.php

```diff
@@ -111,20 +111,31 @@ public function processResult(
         $result = $this->paymentResponseHandlerResult->createFromPaymentResponse($paymentResponse);
 
         if ('RedirectShopper' === $result->getResultCode()) {
-            // Validate 3DS1 Post parameters
+            // Validate 3DS1 parameters
             // Get MD and PaRes to be validated
-            $md = $request->request->get('MD');
-            $paRes = $request->request->get('PaRes');
+            $md = $request->query->get(self::MD);
+            $paRes = $request->query->get(self::PA_RES);
+            $redirectResult = $request->query->get(self::REDIRECT_RESULT);
+            $payload = $request->query->get(self::PAYLOAD);
 
-            if (empty($md) || empty($paRes)) {
-                throw new PaymentException('MD and/or PaRes parameter is missing from the redirect request');
-            }
+            $details = [];
 
             // Construct the details object for the paymentDetails request
-            $details = [
-                'MD' => $md,
-                'PaRes' => $paRes
-            ];
+            if (!empty($md)) {
+                $details[self::MD] = $md;
+            }
+
+            if (!empty($paRes)) {
+                $details[self::PA_RES] = $paRes;
+            }
+
+            if (!empty($redirectResult)) {
+                $details[self::REDIRECT_RESULT] = $redirectResult;
+            }
+
+            if (!empty($payload)) {
+                $details[self::PAYLOAD] = $payload;
+            }
```

Comment:
let's create a ticket to refactor this handler. I think we can pass the full query without parsing the keys one by one. We gonna need to have a list of approved keys like with the state data validation.

Reply:
PW-3123